src/pages/playground/Playground.tsx

```diff
@@ -445,63 +319,33 @@ const Playground: React.FC<PlaygroundProps> = props => {
 
   const hash = isSicpEditor ? props.initialEditorValueHash : location.hash;
   const { uuid } = useParams<{ uuid: string }>();
-  const config = useUrlEncoder();
-  const tokens = useTypedSelector((state: OverallState) => ({
-    accessToken: state.session.accessToken,
-    refreshToken: state.session.refreshToken
-  }));
-
-  const handleURL = useCallback(
-    async (uuid: string | undefined) => {
-      if (uuid !== undefined) {
-        const resp = await requestToShareProgram(`shared_programs/${uuid}`, 'GET', {
-          ...tokens
-        });
-        if (!resp) {
-          return showWarningMessage('Invalid share program link! ');
-        }
-        const respJson = await resp.json();
-        const res: ShareLinkState = new ShareLinkStateDecoder(respJson).decodeWith(
-          new JsonDecoderDelegate()
-        );
-        resetConfig(
-          res,
-          { handleChangeExecTime, handleChapterSelect },
-          workspaceLocation,
-          dispatch,
-          fileSystem
-        );
-        return;
-      } else {
-        const config = new ShareLinkStateDecoder(location.hash).decodeWith(
-          new UrlParamsDecoderDelegate()
-        );
-        resetConfig(
-          config,
-          { handleChangeExecTime, handleChapterSelect },
-          workspaceLocation,
-          dispatch,
-          fileSystem
-        );
-        return;
-      }
-    },
-    // disabled eslint here since tokens are checked separately, checking single object cause infinite rerender.
-    // eslint-disable-next-line
-    [
-      dispatch,
-      fileSystem,
-      handleChangeExecTime,
-      handleChapterSelect,
-      location.hash,
-      workspaceLocation,
-      tokens.accessToken,
-      tokens.refreshToken
-    ]
-  );
 
   useEffect(() => {
-    if (!hash && uuid === undefined) {
+    const getPlaygroundConfigurationFromHash = async (hash: string): Promise<ShareLinkState> =>
+      new ShareLinkStateDecoder(hash).decodeWith(new UrlParamsDecoderDelegate(), workspaceLocation);
```

Comment:
Does this really need to be async? Since we're not awaiting anything, can we just return a resolved promise?

Reply:
Same way of expressing the same idea